src/components/GrowlNotification/index.js

```diff
@@ -55,65 +43,73 @@ class GrowlNotification extends Component {
      * @param {String} type
      * @param {Number} duration
      */
-    show(bodyText, type, duration) {
-        this.setState(
-            {
-                bodyText,
-                type,
-            },
-            () => {
-                this.fling(0);
-                setTimeout(() => {
-                    this.fling(INACTIVE_POSITION_Y);
-                }, duration);
-            },
-        );
-    }
+    // eslint-disable-next-line no-unused-vars
+    const show = useCallback((text, growlType, growlDuration) => {
+        setBodyText(text);
+        setType(growlType);
+        setDuration(growlDuration);
+    }, []);
 
     /**
      * Animate growl notification
      *
      * @param {Number} val
      */
-    fling(val = INACTIVE_POSITION_Y) {
-        Animated.spring(this.state.translateY, {
+    const fling = useCallback((val = INACTIVE_POSITION_Y) => {
+        Animated.spring(translateY, {
             toValue: val,
             duration: 80,
             useNativeDriver: true,
         }).start();
-    }
+    }, []);
 
-    render() {
-        return (
-            <FlingGestureHandler
-                direction={Directions.UP}
-                onHandlerStateChange={({nativeEvent}) => {
-                    if (nativeEvent.state !== State.ACTIVE) {
-                        return;
-                    }
+    useEffect(() => {
+        Growl.setIsReady();
+    }, []);
+
+    useEffect(() => {
+        if (!duration) {
+            return;
+        }
+
+        fling(0);
+        setTimeout(() => {
+            fling(INACTIVE_POSITION_Y);
```

Comment:
As `INACTIVE_POSITION_Y` is also a default value for this function, can't we just `fling()` ?

Reply:
Yes, make sense!

Reply:
resolved